### PR TITLE
rest: Expose block filters follow-ups

### PR DESCRIPTION
**Pull request by dergoegge**
This PR addresses unresolved review comments from [#17631](https://github.com/bitcoin/bitcoin/pull/17631).
This includes: 
* various style fix-ups
* returning a more verbose error message for invalid header counts
* removing superfluous rpc serializations flags for block filters
* improving the test to include comparing the block filters returned from the rest  with the ones returned from the `getblockfilter` RPC.

### CODE REVIEW

File: src/rest.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ static bool rest_headers(const std::any& context,
 
     const auto parsed_count{ToIntegral<size_t>(path[0])};
     if (!parsed_count.has_value() || *parsed_count < 1 || *parsed_count > MAX_REST_HEADERS_RESULTS) {
-        return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Header count out of acceptable range (1-%u): %s",  MAX_REST_HEADERS_RESULTS, path[0]));
+        return RESTERR(req, HTTP_BAD_REQUEST, strprintf("Header count is invalid or out of acceptable range (1-%u): %s", MAX_REST_HEADERS_RESULTS, path[0]));
```
Comment (MarcoFalke) [inline src/rest.cpp:198]: first commit:

Changing the error message is a user-visible behavior change, not a refactor.

File: src/rest.cpp [hunk 2]
```diff
@@ -414,7 +413,7 @@ static bool rest_filter_header(const std::any& context, HTTPRequest* req, const
 
     switch (rf) {
     case RetFormat::BINARY: {
-        CDataStream ssHeader(SER_NETWORK, PROTOCOL_VERSION | RPCSerializationFlags());
+        CDataStream ssHeader(SER_NETWORK, PROTOCOL_VERSION);
```
Comment (MarcoFalke) [inline src/rest.cpp:416]:
```suggestion
        CDataStream ssHeader{SER_NETWORK, PROTOCOL_VERSION};
```

Second commit: nit:

While touching this line, could use C++11 init?

**Review by maflcko — COMMENTED** [review]
Concept ACK. For reference, I've also opened https://github.com/bitcoin/bitcoin/pull/23599 to clean up the RPCTxSerializationFlags function.

I think the pull title and description can be modified to be self-contained. That is, following links is not needed to understand the changes.

File: src/rest.cpp (hunk 1 above)
Comment (dergoegge) [reply src/rest.cpp:198]: I split this into a separate commit

**Review by brunoerg — APPROVED** [review]
tACK 4523d28b6bd550bf9f9c724f9aa90c3d2d4ebe3b

Style fix-ups and the new error message seem ok for me, I ran the test and it passed successfully.

**Comment by fanquake** [comment]
cc @stickies-v @jnewbery

**Comment by jnewbery** [comment]
ACK 4523d28b6bd550bf9f9c724f9aa90c3d2d4ebe3b

Thanks for picking this up!

**Pull request merged**